tf-psa-crypto/core/psa_crypto.c

```diff
@@ -8178,12 +8180,44 @@ psa_status_t psa_generate_key_iop_setup(
 
 psa_status_t psa_generate_key_iop_complete(
     psa_generate_key_iop_t *operation,
-    psa_key_id_t *key)
+    mbedtls_svc_key_id_t *key)
 {
+#if defined(MBEDTLS_ECP_RESTARTABLE)
+    psa_status_t status;
+    uint8_t key_data[PSA_KEY_EXPORT_ECC_KEY_PAIR_MAX_SIZE(PSA_VENDOR_ECC_MAX_CURVE_BITS)+1] = { 0 };
```

Comment:
Sorry, I didn't notice this before, but the `+1` here looks like a missing guard issue to me...
The commit that added this says `Prevent a warning in case PSA_VENDOR_ECC_MAX_CURVE_BITS is set to 0` and I suspect this was generated in builds where no `PSA_WANT_ECC_xxx` was defined, right? If that's the case then do we _really_ need to support  interruptible EC key generation if no EC curve is supported in PSA?
I think guards should be:
- `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE` in `psa_crypto.c` along side `defined(MBEDTLS_ECP_RESTARTABLE)` in `psa_generate_key_iop_setup`, `psa_generate_key_iop_complete`, `psa_generate_key_iop_abort`.
- `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_GENERATE` in `psa_crypto_ecp.c` to guard new interruptible functions for key pair generation.

Wdyt?

Reply:
I agree that the guards eventually should look something like that.

However, guarding this code with PSA_WANT defines is out of scope for this PR and is tracked here: https://github.com/Mbed-TLS/mbedtls/issues/7029

(The driver despatch code is also out of scope and ideally we would add the PSA_WANT defines once we have that as well.)

Reply:
#7029 is about having a PSA equivalent of `MBEDTLS_ECP_RESTARTABLE` (which may or may not be separate macros for each algorithm, depending on whether we minimize conceptual complexity or practical complexity in this particular case). Here the code is already guarded by `MBEDTLS_ECP_RESTARTABLE` so it's a different problem.

At a quick glance at the discussion, and not having read everytihing: maybe we should add something in `check_config.h` to prevent configurations with `MBEDTLS_ECP_RESTARTABLE` enabled but not some core of ECC, in particular at least one restartable curve?